Persistence/Repositories/BankAccountRepository.cs

```diff
@@ -1,49 +1,43 @@
-using System;
-using System.Collections.Generic;
-using System.Threading;
-using System.Threading.Tasks;
-
 using Domain;
 using Domain.Interfaces;
 
 using Microsoft.EntityFrameworkCore;
 
-namespace Persistence.Repositories
+namespace Persistence.Repositories;
+
+public class BankAccountRepository : IBankAccountRepository
 {
-    public class BankAccountRepository : IBankAccountRepository
+    private readonly HaSpManContext _context;
+
+    public BankAccountRepository(IDbContextFactory<HaSpManContext> contextFactory)
+    {
+        _context = contextFactory.CreateDbContext();
+    }
+
+    public void Add(BankAccount account)
+    {
+        _context.BankAccounts.Add(account);
+    }
+
+    public async Task<IEnumerable<BankAccount>> GetAllAsync(CancellationToken ct)
+    {
+        return await _context.BankAccounts.ToListAsync(ct);
+    }
+
+    public async Task<BankAccount?> GetByIdAsync(Guid id, CancellationToken ct)
```

Comment:
Why was this made nullable? A `GetById` method should always return a non-null value, or throw an exception if the requested resource could not be found